apps/mobile/src/libs/components/progress-bar/progress-bar.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import { Dimensions } from "react-native";
+import { default as RNProgressBar } from "react-native-progress-step-bar";
+
+import { LinearGradient, View } from "~/libs/components/components";
+import { GradientColor } from "~/libs/enums/enums";
+import { useMemo } from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
+
+import { SCREEN_WIDTH_MULTIPLIER } from "./constants/constants";
+import { styles } from "./styles";
+
+type Properties = {
+	currentItemIndex: number;
+	totalItemsAmount: number;
+};
+
+const SCREEN_WIDTH: number = Dimensions.get("window").width;
+const PROGRESS_BAR_WIDTH = SCREEN_WIDTH * SCREEN_WIDTH_MULTIPLIER;
+const ZERO = 0;
+const ONE = 1;
+const DOT_DIAMETER = 12;
+const BAR_HEIGHT = 2;
+const STEP_ANIMATION_DURATION = 1000;
+
+const ProgressBar: React.FC<Properties> = ({
+	currentItemIndex,
+	totalItemsAmount,
+}) => {
+	const FilledBar = useMemo(
+		() => (
+			<LinearGradient
+				colors={[...GradientColor.BLUE]}
+				end={{ x: ONE, y: ZERO }}
+				start={{ x: ZERO, y: ZERO }}
```

Comment:
Let's move these objects to separate variables gradientStartCoordinates and gradientEndCoordinates